### ISSUE TITLE
[3.0][Enhancement Request] Hybrid support

### ISSUE THREAD

**Issue by t2ym** [vaadin/vaadin-grid#887]
Is it possible to support hybrid dependencies as follows like the standard Polymer elements such as [`paper-input`](https://github.com/PolymerElements/paper-input/blob/master/bower.json)?

```diff
diff --git a/bower.json b/bower.json
index d176bcc..7027c07 100644
--- a/bower.json
+++ b/bower.json
@@ -47,11 +47,11 @@
     "app-localize-behavior": "^v2.0.0"
   },
   "dependencies": {
-    "polymer": "^2.0.0",
-    "iron-resizable-behavior": "PolymerElements/iron-resizable-behavior#^v2.0.0",
-    "iron-scroll-target-behavior": "PolymerElements/iron-scroll-target-behavior#^v2.0.0",
-    "iron-a11y-keys-behavior": "^v2.0.0",
-    "iron-a11y-announcer": "^v2.0.0"
+    "polymer": "Polymer/polymer#1.9 - 2",
+    "iron-resizable-behavior": "PolymerElements/iron-resizable-behavior#1 - 2",
+    "iron-scroll-target-behavior": "PolymerElements/iron-scroll-target-behavior#1 - 2",
+    "iron-a11y-keys-behavior": "PolymerElements/iron-a11y-keys-behavior#1 - 2",
+    "iron-a11y-announcer": "PolymerElements/iron-a11y-announcer#1 - 2"
   },
   "variants": {
     "1.x": {
```

**Comment by samiheikki** [comment]
Hi @t2ym. I discussed with our team and this could be possible for us to do. Have you tried out installing 3.0 on your app? Does bower automatically resolve the conflicted dependencies or are you required to resolve them manually?

**Comment by t2ym** [comment]
Hi @samiheikki. Thank you for the quick and positive response. You can find my element (not an app) with t2ym/vaadin-grid#hybrid via https://github.com/t2ym/live-localizer/blob/hybrid/bower.json

My element needs a special consideration to support "vaadin-grid#1 - 3" since the element has to be applicable to all apps using any versions of vaadin-grid other than inside of the element. 
 
The delicate issue on the bower resolutions for my special hybrid element (depending on polymer#1.9 - 2, vaadin-grid#1 - 3, polymerfire#0.9 - 2) is that the pseudo-release tag "v3.0.0" is required for vaadin-grid so that bower can have a higher resolution priority in vaadin-grid#3 than #2.  This delicate issue would naturally be resolved by the official release of vaadin-grid#3.

You can find a bower resolution table of my element with target apps in the follwing pre-release description.
https://github.com/t2ym/live-localizer/releases/tag/0.0.69-hybrid.6

For apps directly using vaadin-grid#v3.0.0-xxx, "t2ym/vaadin-grid#v3.0.0-alpha1-hybrid" should work expectedly with either Polymer 1 or 2 and with hybrid and non-hybrid versions of the standard Polymer elements.

**Comment by web-padawan** [comment]
Fixed in #1059 and released in v3.1.0
